include/lbann/utils/exception.hpp

```diff
@@ -46,6 +46,9 @@
     throw lbann::exception(ss_LBANN_ERROR.str());               \
   } while (0)
 
+#define LBANN_ERROR_STR(...)                    \
+  LBANN_ERROR(build_string(__VA_ARGS__))
```

Comment:
I think we should make this the default behavior for `LBANN_ERROR`.

Reply:
I agree, but I didn't want to worry about pushing the change through all of LBANN. Looking at it now, though, I think the result of just making `LBANN_ERROR` have this behavior would just have been a bunch of extra string copies that really don't matter because it's about to throw an exception from which, realistically, LBANN will never recover (though we should move away from that assumption in `liblbann` -- a library's job is just to report errors; it's an application's job to decide to ignore them or attempt recovery).